Lib/test/test_importlib/test_path.py

```diff
@@ -26,6 +27,17 @@ def test_reading(self):
 class PathDiskTests(PathTests, unittest.TestCase):
     data = data01
 
+    def test_package_spec_origin_is_None(self):
+        import pydoc_data
+        spec = pydoc_data.__spec__
+        # Emulate importing from non-file source by setting spec.origin = None.
+        # Barge past path's sanity checks by ensuring spec.loader.is_resource
+        # returns False.
+        with swap_attr(spec, "origin", None), \
+            swap_attr(spec.loader, "is_resource", lambda *args: False), \
```

Comment:
This part of the test feels a little brittle, relying on other implementation details. Another reason to build a fixture that represents the (previously-) unmet expectation.